integreat_cms/static/src/js/unsaved-warning.ts

```diff
@@ -1,40 +1,101 @@
 /**
  * This file contains a function to warn the user when they leave a content without saving
  */
+import { editors, isActuallyDirty } from "./forms/tinymce-init";
 
-let dirty = false;
+const takeFormSnapshot = (form: HTMLFormElement) => new FormData(form);
 
-window.addEventListener("beforeunload", (event) => {
-    // trigger only when something is edited and no submit/save button clicked
-    if (dirty) {
-        event.preventDefault();
-        /* eslint-disable-next-line no-param-reassign */
-        event.returnValue = "This content is not saved. Would you leave the page?";
+const equalSnapshots = (a: FormData, b: FormData) => {
+    // We could inplement a proper comparison of nested arrays
+    // …ooor we could just take the hit and stringify everything
+    // to JSON, saving us maintenance on all these code lines
+    if (a == null || b == null) {
+        return false;
     }
-});
+    const withoutExcluded = (iterator: IterableIterator<[string, FormDataEntryValue]>) =>
+        Array.from(iterator).filter((item) => {
+            const target = document.querySelector(`[name="${item[0]}"]`);
+            return !target.hasAttribute("data-unsaved-warning-exclude");
+        });
+    const aJSON = JSON.stringify(withoutExcluded(a.entries()));
+    const bJSON = JSON.stringify(withoutExcluded(b.entries()));
+    return aJSON === bJSON;
+};
 
-window.addEventListener("load", () => {
-    const form = document.querySelector("[data-unsaved-warning]");
-    // checks whether the user typed something in the content
-    form?.addEventListener("input", (event) => {
-        // does not set the dirty flag, if the event target is explicitly excluded
-        const target = event.target as HTMLElement;
-        if (target.hasAttribute("data-unsaved-warning-exclude")) {
-            return;
+const originalTitle = document.title;
+
+let snapshotCandidate: FormData | null = null;
+let savedSnapshot: FormData | null = null;
+
+const isDirty = (form: HTMLFormElement) => {
+    const now = takeFormSnapshot(form);
+    if (!equalSnapshots(now, savedSnapshot)) {
+        return true;
+    }
+    // Are there any unsaved changes in the editors?
+    for (const editor of editors) {
+        if (isActuallyDirty(editor)) {
+            return true;
         }
-        if (!dirty) {
-            console.debug("editing detected, enabled beforeunload warning");
+    }
+    return false;
```

Comment:
I could reproduce the same problem that @seluianova mentioned, were the modified input would not get submitted sometimes, and could not figure out how exactly it is caused. However, commenting out these lines fixes the problem for me somehow :shrug:

Reply:
Please see whether you can still reproduce it or whether my attempt to fix it was successful

Reply:
@PeterNerlich I still can reproduce it